tensorboard/components/tf_storage/storage.ts

```diff
@@ -164,10 +190,13 @@ function makeBindings<T>(fromString: (string) => T, toString: (T) => string): {
           }
         }
       };
+      if (!fullOptions.useLocalStorage) {
+        // Update it when the hashchanges.
+        window.addEventListener('hashchange', setComponentValue);
+      }
       // Set the value on the property.
       setComponentValue();
-      // Update it when the hashchanges.
-      window.addEventListener('hashchange', setComponentValue);
+      return this[fullOptions.polymerProperty];
```

Comment:
Huh, so before the initializer was returning undefined and somehow still working?  Whereas now we are actually returning the current value of the property as what to "initialize" it with... even though it was really the earlier logic that initialized it.

I guess it structurally works, but it would be a bit cleaner IMO to separate it out so that the hashchange listener function is only invoked as a listener and there logic to do the actual initialization does that by returning the value to Polymer.  I think if the useLocalStorage logic is also extracted out as discussed then it's not much extra code to write it out that way.

Reply:
I think the code now reads a little better and kinda remove a need to do that. Please take a look.